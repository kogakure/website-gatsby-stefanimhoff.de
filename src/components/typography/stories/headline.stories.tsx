import React from 'react';

import { Headline } from '..';

export default {
  title: 'Typography/Headline',
};

export const Default = () => <Headline>Headline</Headline>;

export const NoMargin = () => <Headline noMargin>Headline</Headline>;
