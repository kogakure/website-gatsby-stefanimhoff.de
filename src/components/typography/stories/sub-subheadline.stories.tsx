import React from 'react';

import { SubSubheadline } from '..';

export default {
  title: 'Typography/SubSubheadline',
};

export const Default = () => <SubSubheadline>SubSubheadline</SubSubheadline>;

export const NoMargin = () => (
  <SubSubheadline noMargin>SubSubheadline</SubSubheadline>
);
