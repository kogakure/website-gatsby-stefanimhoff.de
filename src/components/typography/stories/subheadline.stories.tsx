import React from 'react';

import { Subheadline } from '..';

export default {
  title: 'Typography/Subheadline',
};

export const Default = () => <Subheadline>Subheadline</Subheadline>;

export const NoMargin = () => <Subheadline noMargin>Subheadline</Subheadline>;
