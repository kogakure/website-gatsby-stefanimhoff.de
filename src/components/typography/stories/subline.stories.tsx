import React from 'react';

import { Subline } from '..';

export default {
  title: 'Typography/Subline',
};

export const Default = () => <Subline>Subline</Subline>;
