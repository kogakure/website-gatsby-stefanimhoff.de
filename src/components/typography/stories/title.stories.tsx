import React from 'react';

import { Title } from '..';

export default {
  title: 'Typography/Title',
};

export const Default = () => <Title>Title</Title>;

export const NoMargin = () => <Title noMargin>Title</Title>;
