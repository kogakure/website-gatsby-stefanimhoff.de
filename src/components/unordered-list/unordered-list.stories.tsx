import React from 'react';

import { UnorderedList } from '.';

export default {
  component: UnorderedList,
  title: 'Lists/Unordered List',
};

export const Default = () => (
  <UnorderedList>
    <li>
      Vegetables
      <UnorderedList>
        <li>Tomatoe</li>
        <li>Onion</li>
        <li>Potatoe</li>
      </UnorderedList>
    </li>
  </UnorderedList>
);

export const NoMargin = () => (
  <UnorderedList noMargin>
    <li>
      Vegetables
      <UnorderedList>
        <li>Tomatoe</li>
        <li>Onion</li>
        <li>Potatoe</li>
      </UnorderedList>
    </li>
  </UnorderedList>
);
