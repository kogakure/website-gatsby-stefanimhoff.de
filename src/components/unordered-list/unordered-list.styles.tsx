import styled from 'styled-components';

import {
  borderStyle,
  borderWidth,
  fontSize,
  mediaQuery,
  radius,
} from '../../layout/tokens';

const UnorderedListContainer = styled.ul`
  font-size: ${fontSize[3]};
  list-style-type: circle;
  padding-left: 1.2em;

  @media (${mediaQuery.tablet}) {
    padding-left: 0;
  }

  li & {
    padding-left: 2rem;
  }

  &.contains-task-list {
    list-style-type: none;
  }

  @supports (appearance: none) {
    input[type='checkbox'] {
      appearance: none;
      background: transparent;
      border-color: var(--colorTextSoftInverse);
      border-radius: ${radius[25]};
      border-style: ${borderStyle.solid};
      border-width: ${borderWidth[10]};
      cursor: pointer;
      display: inline-block;
      height: 1em;
      margin: 0;
      outline: none;
      position: relative;
      top: 0.3em;
      vertical-align: top;
      width: 1em;

      &:after {
        border-bottom-width: ${borderWidth[15]};
        border-color: var(--colorText);
        border-left-color: transparent;
        border-left-width: 0;
        border-right-width: ${borderWidth[15]};
        border-style: ${borderStyle.solid};
        border-top-color: transparent;
        border-top-width: 0;
        content: '';
        display: block;
        height: 0.5em;
        left: 0.25em;
        opacity: var(--opacity, 0);
        position: absolute;
        top: 0.1em;
        transform: rotate(var(--rotate, 0deg));
        width: 0.3em;
      }

      &:checked {
        --rotate: 43deg;
        --opacity: 1;
      }

      &:disabled {
        cursor: not-allowed;
      }
    }
  }
`;

export const Styled = {
  UnorderedListContainer,
};
