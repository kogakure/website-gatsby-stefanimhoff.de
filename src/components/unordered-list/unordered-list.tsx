import * as React from 'react';

import { Styled } from './unordered-list.styles';

export const UnorderedList: React.FC = ({ children, ...props }) => (
  <Styled.UnorderedListContainer {...props}>
    {children}
  </Styled.UnorderedListContainer>
);
