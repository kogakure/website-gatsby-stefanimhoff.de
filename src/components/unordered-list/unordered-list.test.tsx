import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../utils/test-utils';

import { Default } from './unordered-list.stories';

describe('UnorderedList', () => {
  test('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('is accessible', async () => {
    const { container } = render(<Default />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
