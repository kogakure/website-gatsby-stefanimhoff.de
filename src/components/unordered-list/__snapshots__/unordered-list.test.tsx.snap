// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedList renders correctly 1`] = `
.c0 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
  list-style-type: circle;
  padding-left: 0;
}

li .c0 {
  padding-left: 2rem;
}

.c0.contains-task-list {
  list-style-type: none;
}

@supports (-webkit-appearance:none) or (-moz-appearance:none) or (appearance:none) {
  .c0 input[type='checkbox'] {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    background: transparent;
    border-color: var(--colorTextSoftInverse);
    border-radius: 25%;
    border-style: solid;
    border-width: 0.1em;
    cursor: pointer;
    display: inline-block;
    height: 1em;
    margin: 0;
    outline: none;
    position: relative;
    top: 0.3em;
    vertical-align: top;
    width: 1em;
  }

  .c0 input[type='checkbox']:after {
    border-bottom-width: 0.15em;
    border-color: var(--colorText);
    border-left-color: transparent;
    border-left-width: 0;
    border-right-width: 0.15em;
    border-style: solid;
    border-top-color: transparent;
    border-top-width: 0;
    content: '';
    display: block;
    height: 0.5em;
    left: 0.25em;
    opacity: var(--opacity,0);
    position: absolute;
    top: 0.1em;
    -webkit-transform: rotate(var(--rotate,0deg));
    -ms-transform: rotate(var(--rotate,0deg));
    transform: rotate(var(--rotate,0deg));
    width: 0.3em;
  }

  .c0 input[type='checkbox']:checked {
    --rotate: 43deg;
    --opacity: 1;
  }

  .c0 input[type='checkbox']:disabled {
    cursor: not-allowed;
  }
}

<ul
  class="c0"
>
  <li>
    Vegetables
    <ul
      class="c0"
    >
      <li>
        Tomatoe
      </li>
      <li>
        Onion
      </li>
      <li>
        Potatoe
      </li>
    </ul>
  </li>
</ul>
`;
